packages/rspack/src/utils/resolve-user-defined-rspack-config.ts

```diff
@@ -18,7 +18,7 @@ export function resolveUserDefinedRspackConfig(
   // Don't transpile non-TS files. This prevents workspaces libs from being registered via tsconfig-paths.
   // There's an issue here with Nx workspace where loading plugins from source (via tsconfig-paths) can lead to errors.
   if (!/\.(ts|mts|cts)$/.test(path)) {
-    return require(path);
+    return import(path);
```

Comment:
This changes the function to return a Promise in _some_ circumstances.

That's going to impact how this function is used in `config.ts` and `plugin.ts`.